lib/js/components/OverlayHeader/OverlayHeader.vue

```diff
@@ -0,0 +1,303 @@
+<template>
+	<div
+		class="ds-overlayHeader"
+		:class="{
+			'-borderNeutral': borderColor === OVERLAY_HEADER_BORDER_COLORS.NEUTRAL,
+			'-borderSuccess': borderColor === OVERLAY_HEADER_BORDER_COLORS.SUCCESS,
+			'-borderFail': borderColor === OVERLAY_HEADER_BORDER_COLORS.FAIL,
+			'-borderWarning': borderColor === OVERLAY_HEADER_BORDER_COLORS.WARNING,
+		}"
+	>
+		<div v-if="$slots.accessory" class="ds-overlayHeader__accessory">
+			<slot name="accessory" />
+		</div>
+		<div class="ds-overlayHeader__content">
+			<div v-if="eyebrowText || $slots.eyebrowAccessory" class="ds-overlayHeader__eyebrow">
+				<div v-if="eyebrowText" class="ds-overlayHeader__eyebrowText" :title="eyebrowText">
+					{{ eyebrowText }}
+				</div>
+				<div v-if="$slots.eyebrowAccessory" class="ds-overlayHeader__eyebrowAccessory">
+					<slot name="eyebrowAccessory"></slot>
+				</div>
+			</div>
+			<div class="ds-overlayHeader__main">
+				<div v-if="$slots.titleLeading" class="ds-overlayHeader__titleLeading">
+					<slot name="titleLeading" />
+				</div>
+				<div
+					class="ds-overlayHeader__titleWrapper"
+					:title="title"
+					:class="{ '-interactive': isTitleInteractive }"
+					@click="onTitleClick"
+				>
+					<div class="ds-overlayHeader__title -desktop">
+						{{ title }}
+					</div>
+					<div class="ds-overlayHeader__title -mobile">
+						{{ shortTitle || title }}
+					</div>
+				</div>
+				<div v-if="$slots.titleTrailing" class="ds-overlayHeader__titleTrailingWrapper">
+					<div class="ds-overlayHeader__titleTrailing">
+						<slot name="titleTrailing" />
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<template v-if="$slots.actions">
+			<div class="ds-overlayHeader__actions">
+				<slot name="actions" />
+			</div>
+			<ds-divider
+				class="ds-overlayHeader__divider -mobileHidden"
+				:prominence="DIVIDER_PROMINENCES.STRONG"
+				is-vertical
+			/>
+		</template>
+		<template v-if="$scopedSlots.dropdown">
+			<ds-dropdown boundaries-selector="body">
+				<template #reference>
+					<icon-button
+						:icon="ICONS.FA_ELLIPSIS_VERTICAL"
+						:size="ICON_BUTTON_SIZES.MEDIUM"
+						:color="ICON_BUTTON_COLORS.NEUTRAL"
+					/>
+				</template>
+				<template #default="{ close }">
+					<slot name="dropdown" :close="close" />
+				</template>
+			</ds-dropdown>
+			<ds-divider
+				class="ds-overlayHeader__divider"
+				:prominence="DIVIDER_PROMINENCES.STRONG"
+				is-vertical
+			/>
+		</template>
+		<icon-button
+			:icon="ICONS.FA_XMARK"
+			:size="ICON_BUTTON_SIZES.MEDIUM"
+			:color="ICON_BUTTON_COLORS.NEUTRAL"
+			@click.native="$emit('close')"
+		/>
+	</div>
+</template>
+
+<style scoped lang="scss">
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/media-queries';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/typography/tokens';
+@import '../../../styles/mixins/flex_overflow_mask';
```

Comment:
Why not kebab-case like other files?

Reply:
[`3da9bb8` (#260)](https://github.com/bethinkpl/design-system/pull/260/commits/3da9bb8acdaded82cfa0c106341da0ef46a18ce5)